protocol/communities/manager.go

```diff
@@ -1894,58 +1896,81 @@ func (m *Manager) CreateHistoryArchiveTorrent(communityID types.HexBytes, msgs [
 			continue
 		}
 
-		wakuMessageArchive := m.createWakuMessageArchive(from, to, messages, topicsAsByteArrays)
-		encodedArchive, err := proto.Marshal(wakuMessageArchive)
-		if err != nil {
-			return archiveIDs, err
+		// Not only do we partition messages, we also chunk them
+		// roughly by size, such that each chunk will not exceed a given
+		// size and archive data doesn't get too big
+		messageChunks := make([][]types.Message, 0)
+		currentChunkSize := 0
+		currentChunk := make([]types.Message, 0)
+		for i, msg := range messages {
+			msgSize := len(msg.Payload) + len(msg.Sig)
+			if currentChunkSize+msgSize > maxArchiveSizeInBytes && len(currentChunk) > 0 {
+				messageChunks = append(messageChunks, currentChunk)
+				currentChunk = make([]types.Message, 0)
+				currentChunkSize = 0
+			}
+			currentChunk = append(currentChunk, msg)
+			currentChunkSize = currentChunkSize + msgSize
+			if i == len(messages)-1 {
+				messageChunks = append(messageChunks, currentChunk)
```

Comment:
It looks to me that if you move `messageChunks = append(messageChunks, currentChunk)` after the loop, this check `if i == len(messages)-1 {` is not needed; what do you think?

Reply:
Good catch, updated!